pkg/instaclustr/client.go

```diff
@@ -2215,6 +2215,31 @@ func (c *Client) UpdateClusterSettings(clusterID string, settings *models.Cluste
 	return nil
 }
 
+func (c *Client) CheckIfAWSEndpointServicePrincipalExists(id string) (bool, error) {
```

Comment:
Let's rename this method to GetAWSEndpointServicePrincipal and will return the service principal entity itself and an error. We can check the existence of the principal by using `errors.Is(err, instaclustr.NotFound)` func also

Reply:
Fixed